### PR TITLE
Lesson 7 Studio

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: src/chapters/interfaces-and-polymorphism/studio.rst [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+   for this project and fork it to create a copy under your own GitHub account.
+#. Open up the solution in Visual Studio to get started.
+
+Disc Project Overview
+----------------------
+
+In this activity, you will pair up to design a new custom interface.
+
+Take a look at the prepared ``CD`` and ``DVD`` classes in the starter
+code---only their signatures are present. You will add fields and methods by
```
Comment (CarlyLanglois) [inline src/chapters/interfaces-and-polymorphism/studio.rst]: is a class definition called a signature? i've only seen this term used for methods before

File: src/chapters/interfaces-and-polymorphism/studio.rst [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+extending a base class and implementing your interface. Before you do this,
+however, you need to consider what code to place in the interface vs. the
+abstract class vs. each specific class.
+
+Just like you did with your Restaurant Menu studios, you will begin with pen
```
Comment (CarlyLanglois) [inline src/chapters/interfaces-and-polymorphism/studio.rst]: I'd a link back to the first studio here

File: src/chapters/interfaces-and-polymorphism/studio.rst [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+------------------------------------------
+
+Next, you and your partner need to decide which behaviors go into the
+interface vs. an abstract class.
+
+Behaviors added to an interface do NOT have to be identical for every object
+that implements the interface. The key is that the methods require no
+object-specific data. The methods can accept parameters, but these should
+represent generic values like ``String searchTerm`` rather than field data
+like ``DVD.title``.
```
Comment (CarlyLanglois) [inline src/chapters/interfaces-and-polymorphism/studio.rst]: how do you feel about mentioning these terms with the code syntax? It's not clear to me what a searchTerm would be in this app design also - can you elaborate on this one?

File: src/chapters/interfaces-and-polymorphism/studio.rst [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+For example, even though CDs and DVDs both spin, they do so at different
+speeds. Their behavior (spinning) is the same, but the details of that
+behavior vary. Thus, a method like ``SpinDisc()`` fits well into an interface.
+
+#. For the other behaviors you and your partner sorted, decide which ones
+   belong in the interface.
+#. Write down details about how these behaviors differ for CDs and DVDs. This
+   guides your thinking about how each class should override the methods.
```
Comment (CarlyLanglois) [inline src/chapters/interfaces-and-polymorphism/studio.rst]: implement the methods

File: src/chapters/interfaces-and-polymorphism/studio.rst [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+   b. Do any of the members represent *constants*?
+
+Code the Interface
+------------------
+
+Now that your team has completed the before-coding thinking, it's time to
+start building!
+
+#. Add an ``IOpticalDisc`` file for the interface. Refer back to
+   :ref:`Creating an Interface <create-an-interface>` if you need a quick
```
Comment (CarlyLanglois) [inline src/chapters/interfaces-and-polymorphism/studio.rst:93]: this should be in master now fyi

File: src/chapters/interfaces-and-polymorphism/studio.rst [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+Bonus Missions: Other Discs & Disks?
+-------------------------------------
+
+#. CDs and DVDs are not the only flat, circular media that have been used to
+   store and return data.
+
+   a. Add classes for vinyl records (LPs) and floppy disks.
+   b. Does your interface apply to all four classes? If so, implement the
+      interface into the LP and floppy classes, and complete the appropriate
```
Comment (CarlyLanglois) [inline src/chapters/interfaces-and-polymorphism/studio.rst]: implement the interface and its methods in the LP and floppy classes.

File: src/chapters/interfaces-and-polymorphism/studio.rst [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+Set up a local copy of the project:
+
+#. Visit the `repository page <https://github.com/LaunchCodeEducation/csharp-web-dev-lsn7interfaces-studio>`__
+   for this project and fork it to create a copy under your own GitHub account.
+#. Open up the solution in Visual Studio to get started.
+
+Disc Project Overview
+----------------------
+
+In this activity, you will pair up to design a new custom interface.
```
Comment (CarlyLanglois) [inline src/chapters/interfaces-and-polymorphism/studio.rst]: and an abstract class

**Review by CarlyLanglois — APPROVED** [review]
👍 - and the solution is great too!

File: src/chapters/interfaces-and-polymorphism/studio.rst (hunk 2 above)
Comment (gildedgardenia) [reply src/chapters/interfaces-and-polymorphism/studio.rst]: Done!

File: src/chapters/interfaces-and-polymorphism/studio.rst (hunk 4 above)
Comment (gildedgardenia) [reply src/chapters/interfaces-and-polymorphism/studio.rst]: Done!

File: src/chapters/interfaces-and-polymorphism/studio.rst (hunk 6 above)
Comment (gildedgardenia) [reply src/chapters/interfaces-and-polymorphism/studio.rst]: Done!

File: src/chapters/interfaces-and-polymorphism/studio.rst (hunk 7 above)
Comment (gildedgardenia) [reply src/chapters/interfaces-and-polymorphism/studio.rst]: Done!

File: src/chapters/interfaces-and-polymorphism/studio.rst (hunk 1 above)
Comment (gildedgardenia) [reply src/chapters/interfaces-and-polymorphism/studio.rst]: Fixed the wording!

File: src/chapters/interfaces-and-polymorphism/studio.rst (hunk 3 above)
Comment (gildedgardenia) [reply src/chapters/interfaces-and-polymorphism/studio.rst]: They were just examples. I removed the syntax highlighting so it doesn't seem like those are fields that the app has to contain.

File: src/chapters/interfaces-and-polymorphism/studio.rst (hunk 5 above)
Comment (gildedgardenia) [reply src/chapters/interfaces-and-polymorphism/studio.rst:93]: Got it!

**Pull request merged**